### DISCUSSION TITLE
Troubleshooting Default Value Display  in Custom Resource Page

### DISCUSSION THREAD

**Discussion by AsutoshBaral** [filamentphp/filament#9376] [General]
I have built a custom resource page, and inside it, I have a 'mount' function. However, the default value doesn't appear when I click on the 'Send Email' button and  I want to be able to edit the page after receiving the default value and save the edited data in the database. How can I do this?

**Comment by leandrocfe** [comment]
Share the code please

**Reply by AsutoshBaral** [reply]
class ResponseRequest extends EditRecord
{

    protected static string $resource = ResponseResource::class;

    protected static string $layout = 'components.response-request.layouts.index';

    protected function authorizeAccess(): void
    {
    }

    public function mount(int | string $record = null): void
    {
      
        try {
            $data = JWT::decode(app('request')->input('token'), ['header' => true]);
        } catch (\Throwable $th) {
            abort(403);
        }
        $response = Response::where([['id', $data['response_id']], ['status', ResponseStatus::DRAFT]])->first();
        if (!$response) {
            abort(404);
        }
        parent::mount($response->id); 
    }

    protected function getHeaderActions(): array
    {
        return [
            //
        ];
    }

    protected function afterSave(): void
    {
        $this->record->status = ResponseStatus::SUBMITTED();
        $this->record->save();
    }

    public function form(Form $form): Form
    {
        $form = static::$resource::form($form, true);
        return $form;
    }
}

**Reply by AsutoshBaral** [reply]
How to fix it ??
Please help me to solve this issue. Thank you.

**Comment by sinnbeck** [comment]
What 'Send Email' button?

And I am personally not sure what you are trying to do here?

Out of curiosity. Is this a school assignment? I notice your code is almost identical to this
https://github.com/filamentphp/filament/discussions/9383

**Reply by AsutoshBaral** [reply]
My workflow is like this:
I have created a record, and after that, an email is triggered to the respective manager with the same data I have created. The data is initially rendered using the mount() method, and the respective manager can edit the field data and save that data.

It worked perfectly in Filament v2, but now I am facing an issue in Filament v3. In Filament v3, the error I'm facing is that my form data is not showing in the UI. However, when I use the dd method to inspect the response, I can see that all the data is present.

The error I encounter in the UI is as follows:
"Unable to find component: [app.filament.resources.appraisal-response-resource.pages.appraisal-response-request]"

**Reply by sinnbeck** [reply]
Where does `appraisal-response-request` come from ? The word appraisal is not mentioned once in the shared code

**Reply by AsutoshBaral** [reply]
i have fixed that with  protected static string $layout = 'components.appraisal-response-request.layouts.index';
still this error is showing

**Reply by AsutoshBaral** [reply]
Here is my updated code:
class AppraisalResponseRequest extends EditRecord
{

    protected static string $resource = AppraisalResponseResource::class;

    protected static string $layout = 'components.appraisal-response-request.layouts.index';

    public function mount(int | string $record = null): void
    { 
        try {
            $data = JWT::decode(app('request')->input('token'), ['header' => true]);
        } catch (\Throwable $th) {
            abort(403);
        }
        
        $appraisal_response = AppraisalResponse::where([['id', $data['response_id']], ['status', AppraisalResponseStatus::DRAFT]])->first();
        if (!$appraisal_response) {
            abort(404);
        }
        parent::mount($appraisal_response->id);
    }

    protected function getHeaderActions(): array
    {
        return [
            //
        ];
    }

    protected function afterSave(): void
    {
        $this->record->status = AppraisalResponseStatus::SUBMITTED();
        $this->record->save();
    }

    public function form(Form $form): Form
    {
       
        $form = static::$resource::form($form, true);
        return $form;
    }
}

**Reply by shyam0118** [reply]
I am also facing the same issue
How to fix it ?

**Comment by leynnnnnn0** [comment]
For the default value to be displayed on customer resource page.

1st step - Create an array property 
public ?array $data = [];

2nd step - add statePath('data') in form
public function form(\Filament\Forms\Form $form): \Filament\Forms\Form
    {
        return $form->schema([
            TextInput::make('service_job_id')
                ->default('AN-' . random_int(100000, 999999))
                ->disabled()
                ->required()
                ->maxLength(32)
                ->label('Service Job Number')
                ->unique(ServiceJob::class, 'id', ignoreRecord: true),

            TextInput::make('appointment_number')
                ->default($this->record->appointment_number)
                ->disabled(),

            TextInput::make('car_id')
                ->default($this->record->car_id)
                ->disabled()
        ])->statePath('data');

    }
